tests/semver/test_version.py

```diff
@@ -113,18 +113,38 @@ def test_allows():
     assert not v.allows(Version.parse("1.3.3"))
     assert not v.allows(Version.parse("1.2.4"))
     assert not v.allows(Version.parse("1.2.3-dev"))
-    assert not v.allows(Version.parse("1.2.3+build"))
+    assert v.allows(Version.parse("1.2.3+build"))
+    assert v.allows(Version.parse("1.2.3-1"))
+    assert v.allows(Version.parse("1.2.3-1+build"))
+
+
+def test_allows_with_local():
+    v = Version.parse("1.2.3+build.1")
+    assert v.allows(v)
+    assert not v.allows(Version.parse("1.3.3"))
+    assert not v.allows(Version.parse("1.2.3-dev"))
+    assert not v.allows(Version.parse("1.2.3+build.2"))
+    assert v.allows(Version.parse("1.2.3-1"))
+    assert v.allows(Version.parse("1.2.3-1+build.1"))
+
+
+def test_allows_with_post():
+    v = Version.parse("1.2.3-1")
+    assert v.allows(v)
+    assert not v.allows(Version.parse("1.2.3"))
+    assert not v.allows(Version.parse("2.2.3"))
+    assert not v.allows(Version.parse("1.2.3-dev"))
+    assert not v.allows(Version.parse("1.2.3+build.2"))
+    assert v.allows(Version.parse("1.2.3-1+build.1"))
 
 
 def test_allows_all():
     v = Version.parse("1.2.3")
 
     assert v.allows_all(v)
     assert not v.allows_all(Version.parse("0.0.3"))
-    assert not v.allows_all(
```

Comment:
@sdispater from what I can tell this was incorrect here. I am guessing this is the right fix.

Reply:
It is correct though. Version 1.2.3 does not allow all versions in the range 1.1.4 to 1.2.4 and a single version does not allow any version.